controllers/page/PageHandler.inc.php

```diff
@@ -178,6 +146,81 @@ function css($args, $request) {
 				}
 		}
 	}
+	/**
+	 * Setup and assign variables for any templates that want the overall header
+	 * context.
+	 * @param $args array
+	 * @param $request PKPRequest
+	 * @return JSONMessage JSON object
+	 */
+	private function setupHeader($args, $request) {
+
+		$templateMgr = TemplateManager::getManager($request);
+
+		$workingContexts = $this->getWorkingContexts($request);
+		$context = $request->getContext();
+
+		if ($workingContexts && $workingContexts->getCount() > 1) {
+			$dispatcher = $request->getDispatcher();
+			$contextsNameAndUrl = array();
+			while ($workingContext = $workingContexts->next()) {
+				$contextUrl = $dispatcher->url($request, ROUTE_PAGE, $workingContext->getPath());
+				$contextsNameAndUrl[$contextUrl] = $workingContext->getLocalizedName();
+			}
+
+			// Get the current context switcher value. We don´t need to worry about the
+			// value when there is no current context, because then the switcher will not
+			// be visible.
+			$currentContextUrl = null;
+			if ($context) {
+				$currentContextUrl = $dispatcher->url($request, ROUTE_PAGE, $context->getPath());
+			} else {
+				$contextsNameAndUrl = array(__('context.select')) + $contextsNameAndUrl;
+			}
+
+			$templateMgr->assign(array(
+				'currentContextUrl' => $currentContextUrl,
+				'contextsNameAndUrl' => $contextsNameAndUrl,
+				'multipleContexts' => true
+			));
+		} else {
+			$templateMgr->assign('noContextsConfigured', true);
+			if (!$workingContexts) {
+				$templateMgr->assign('notInstalled', true);
+			}
+		}
+
+		// OMP only
```

Comment:
Is this actually OMP-only code? If so, it should go into the OMP repo (e.g. in a subclass).